alonet/panoptic/criterion.py

```diff
@@ -0,0 +1,187 @@
+                self.loss_weights.update({f"loss_DICE_{i}": loss_dice_weight})
+
+    def loss_masks(
+        self, outputs: dict, frames: aloscene.Frame, indices: list, num_boxes: torch.Tensor, **kwargs
+    ) -> Dict:
+        """Compute the losses related to the bounding boxes, the L1 regression loss and the GIoU loss
+
+        Parameters
+        ----------
+        outputs: dict
+            Detr model forward outputs
+        frames: aloscene.Frane
+            Trgat frame with boxes2d and labels
+        indices: list
+            List of tuple with predicted indices and target indices
+        num_boxes: torch.Tensor
+            Number of total target boxes
+
+        Returns
+        -------
+        Dict:
+            DICE and focal losses results
+        """
+        assert frames.names[0] == "B"
+        assert frames.segmentation[0].labels is not None and frames.segmentation[0].labels.encoding == "id"
+
+        losses = {}
+        if num_boxes == 0:
+            return {}
+
+        target_masks = torch.cat(
+            [
+                # Select masks per batch following the `target_indices` from the Hungarian matching
+                masks.as_tensor()[indices[b][1]]
+                for b, masks in enumerate(frames.segmentation)
+            ],
+            dim=0,
+        )
+
+        # Masks filtered in forward. Remove pad
+        pred_masks = torch.cat([masks[: len(idx)] for (_, idx), masks in zip(indices, outputs["pred_masks"])], dim=0)
+        pred_masks = F.interpolate(
+            pred_masks[:, None], size=target_masks.shape[-2:], mode="bilinear", align_corners=False
+        )
+        pred_masks = pred_masks[:, 0].flatten(1)
+        target_masks = target_masks.flatten(1).view(pred_masks.shape)
+
+        # DICE/F-1 loss
+        losses["loss_DICE"] = dice_loss(pred_masks, target_masks, num_boxes)
+
+        # Sigmoid focal loss
+        losses["loss_focal"] = sigmoid_focal_loss(pred_masks, target_masks, num_boxes, **kwargs)
+
+        return losses
+
+    def get_loss(
+        self, loss: str, outputs: dict, frames: aloscene.Frame, indices: list, num_boxes: torch.Tensor, **kwargs
+    ) -> Dict:
+        """Compute a loss given the model outputs, the target frame, the results from the matcher
+        and the number of total boxes accross the devices.
+
+        Parameters
+        ----------
+        loss: str
+            Name of the loss to compute
+        outputs: dict
+            Detr model forward outputs
+        frames: aloscene.Frane
+            Trgat frame with boxes2d and labels
+        indices: list
+            List of tuple with predicted indices and target indices
+        num_boxes: torch.Tensor
+            Number of total target boxes
+
+        Returns
+        -------
+        Dict
+            Losses of the loss procedure.
+        """
+        loss_map = {"labels": self.loss_labels, "boxes": self.loss_boxes, "masks": self.loss_masks}
```

Comment:
Would be better to surcharge instead of inherit this method so that this criterion could be plug & play on different criterion following the same style.

Reply:
@thibo73800  What do you mean?  You suggest update `detr_criterion`, right?